contract-ui/tabs/settings/page.tsx

```diff
@@ -43,7 +50,8 @@ export const CustomContractSettingsTab: React.FC<
     detectedMetadata === "disabled" &&
     detectedPrimarySale === "disabled" &&
     detectedPlatformFees === "disabled" &&
-    detectedRoyalties === "disabled"
+    detectedRoyalties === "disabled" &&
+    detectedAppURI === "disabled"
```

Comment:
repetitive code, store ```disabled``` as a constant value at a single place.
You can remove all these extra variables with the pipe function output, you can utilize the power of ```some``` here.

A lot of repetitive code in react components also. Can't comment on them as you can comment on current changes only.